utility/doc/E2E_FEATURE_LIST.md

```diff
@@ -0,0 +1,121 @@
+# E2E Features <!-- omit in toc -->
+
+- [Legend](#legend)
+- [A. E2E Relay](#a-e2e-relay)
+  - [1. Trustless Relay 🟡](#1-trustless-relay-)
+  - [2. Delegated Relay 🟡](#2-delegated-relay-)
+  - [3. Rate Limited Application 🟢🔴](#3-rate-limited-application-)
```

Comment:
@adshmh Going to open an inline thread per your comment in  https://github.com/pokt-network/pocket/pull/766#pullrequestreview-1440778055

> Too early for me to propose any potential changes, but I do have a question regarding the TestNet vs MainNet on item 3, i.e. Rate Limited Applications: is there enough distinction between the two E2E tests that having the test ready for MainNet does not cover having it on TestNet?

There's no formal definition (written down anywhere) for what we need to have in TestNet vs MainNet so it's a good question. Even between the [specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/utility), the paper on [Relay Mining](https://arxiv.org/abs/2305.10672), and all the other notes we have, there are still open-ended questions w.r.t the protocol which will be answered by the future economics group.

For TestNet, we need to have all the building blocks (i.e. entrypoints) in place to validate that values are changing, communication/messaging (i.e. web2 style) is reliable and we can build upon it. E2E tests should capture behaviour. For MainNet, we'll need to make sure that everything is completely secure (i.e. no cryptographic vulnerabilities), proven (i.e. the algorithms are sound, scalable and on-chain) and we can build upon the tokenomics. In other words, TestNet -> MainNet will be the bigger hurdle.